src/ResponseValidator.ts

```diff
@@ -94,20 +94,18 @@ export class ResponseValidator {
 
         // OpenID Connect Core 1.0 says that id_token is optional in refresh response:
         // https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokenResponse
-        if (response.isOpenId) {
-            if (response.id_token) {
-                this._validateIdTokenAttributes(response, state.id_token);
-                logger.debug("ID Token validated");
-            }
-            else {
-                // if there's no id_token on the response, copy over id_token from original request
-                response.id_token = state.id_token;
-                // and decoded part too
-                response.profile = state.profile;
-            }
+        if (response.id_token) {
```

Comment:
I am wondering if that change goes a bit too far as it now tries to parse id_token even if it is not OpenID connect. Probably an esoteric use case, but maybe it is clearer to only change the reuse id_token case and still require the request to have the scope when actually trying parse the response token.

Reply:
i have adapted the code accordingly. I only check for `response.isOpenId` before `_validateIdTokenAttributes` and `_processClaims`, as we do so in all other places within that file. `_validateIdTokenAttributes` handles the non case directly, where `_processClaims` goes over profile, which is only set when isOpenId was true.